### PR TITLE
Ability to filter with case insensitive tags

### PR DESCRIPTION
**Pull request by haroon-sheikh**
Signed-off-by: Haroon Sheikh <haroon.sheikh@outlook.com>

### CODE REVIEW

File: filter/specItemFilter.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func sanitize(tag string) string {
 	if _, err := strconv.ParseBool(tag); err == nil {
 		return fmt.Sprintf("{%s}", tag)
 	}
-	return tag
+	return strings.ToLower(tag)
```
Comment (zabil) [inline filter/specItemFilter.go:73]: Thanks for the PR @haroon-sheikh can we add a new property to make this a configuration? By default it's case sensitive with the option to turn it off. Worth observing if anyone uses case sensitive tag and making case insensitivity the default option in a future release.

Comment (haroon-sheikh) [reply filter/specItemFilter.go:73]: My only concern is, if we turn it off by default? then its very unlikely to be used by others and we'll never find out if they're using it in a case sensitive way.

Comment (haroon-sheikh) [reply filter/specItemFilter.go]: I've now added a flag around it. Let me know what you think?

File: env/env.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ const (
 	saveExecutionResult = "save_execution_result"
 	// CsvDelimiter holds delimiter used to parse csv files
 	CsvDelimiter                   = "csv_delimiter"
+	allowCaseInSensitiveTags       = "allow_case_insensitive_tags"
```
Comment (zabil) [inline env/env.go:41]: For readabioity better to call this 
```suggestion
	allowCaseSensitiveTags       = "allow_case_sensitive_tags"
```

And set it as false by default.

**Review by zabil — CHANGES_REQUESTED** [review]

File: env/env.go (hunk 2 above)
Comment (haroon-sheikh) [reply env/env.go:41]: I'm confused @zabil. Do we want to enable case sensitivity by default?

Comment (haroon-sheikh) [reply env/env.go]: Done.

File: env/env.go [hunk 3]
```diff
@@ -115,6 +116,7 @@ func loadDefaultEnvVars() {
 	defaultScreenshotDir := filepath.Join(config.ProjectRoot, common.DotGauge, "screenshots")
 	addEnvVar(GaugeScreenshotsDir, defaultScreenshotDir)
 	addEnvVar(gaugeSpecFileExtensions, ".spec, .md")
+	addEnvVar(allowCaseInSensitiveTags, "false")
```
[the remark this answers is not in the document]
Comment (haroon-sheikh) [reply env/env.go]: Done.

File: env/env.go [hunk 4]
```diff
@@ -309,3 +311,8 @@ var GaugeSpecFileExtensions = func() []string {
 	}
 	return allowedExts
 }
+
+// AllowCaseInSensitiveTags determines if the casing is ignored in tags filtering
+var AllowCaseInSensitiveTags = func() bool {
```
[the remark this answers is not in the document]
Comment (haroon-sheikh) [reply env/env.go]: Done.

File: env/env_test.go [hunk 5]
```diff
@@ -39,6 +39,7 @@ func (s *MySuite) TestLoadDefaultEnv(c *C) {
 	defaultScreenshotDir := filepath.Join(config.ProjectRoot, common.DotGauge, "screenshots")
 	c.Assert(os.Getenv("gauge_screenshots_dir"), Equals, defaultScreenshotDir)
 	c.Assert(os.Getenv("gauge_spec_file_extensions"), Equals, ".spec, .md")
+	c.Assert(os.Getenv("allow_case_insensitive_tags"), Equals, "false")
```
[the remark this answers is not in the document]
Comment (haroon-sheikh) [reply env/env_test.go]: Done.

File: filter/specItemFilter.go [hunk 6]
```diff
@@ -70,6 +71,9 @@ func sanitize(tag string) string {
 	if _, err := strconv.ParseBool(tag); err == nil {
 		return fmt.Sprintf("{%s}", tag)
 	}
+	if env.AllowCaseInSensitiveTags() {
```
[the remark this answers is not in the document]
Comment (haroon-sheikh) [reply filter/specItemFilter.go]: Done.

File: filter/specItemFilter_test.go [hunk 7]
```diff
@@ -19,6 +20,15 @@ type MySuite struct{}
 
 var _ = Suite(&MySuite{})
 
+func before()  {
+	os.Clearenv()
+	os.Setenv("allow_case_insensitive_tags", "true")
```
[the remark this answers is not in the document]
Comment (haroon-sheikh) [reply filter/specItemFilter_test.go]: Done.

File: version/version.go [hunk 8]
```diff
@@ -14,7 +14,7 @@ import (
 )
 
 // CurrentGaugeVersion represents the current version of Gauge
-var CurrentGaugeVersion = &Version{1, 3, 1}
+var CurrentGaugeVersion = &Version{1, 3, 2}
```
[the remark this answers is not in the document]
Comment (haroon-sheikh) [reply version/version.go]: Done.

**Review by zabil — APPROVED** [review]

**Comment by chadlwilson** [comment]
Thanks for "fixing" this. This seems to have highlighted that some specs for [GoCD](https://github.com/gocd/gocd) were not actually running as intended due to the previous case sensitivity :'(

**Pull request merged**